### ISSUE TITLE
torch xpu event has ~0.1ms latency, which is too large

### ISSUE THREAD

**Issue by jianyizh** [intel/torch-xpu-ops#3227]
### 🐛 Describe the bug

`use_profiler = True` on b580 get 0.0009ms, `use_profiler = False` on b580 get 0.13ms, xpu overhead is ~120us
`use_profiler = True` on 4080s get 0.0013ms, `use_profiler = False` on b580 get 0.007ms,  cuda overhead is ~6us
```python
import torch
import time
import os
from torch.profiler import profile, ProfilerActivity
from torch.utils.cpp_extension import load

my_neg = torch.neg_

tensor_shape_to_flush_cache = (64, 1024, 1024)
device = torch.accelerator.current_accelerator().type

activities = []
use_profiler = False
if torch.xpu.is_available():
    activities.append(ProfilerActivity.XPU)
else:
    activities.append(ProfilerActivity.CUDA)

def warm_up_l2_cache(x, flag = False):
    if not flag:
        return
    for _ in range(10):
        x.add_(1.)
        x.add_(-1.)
    x.sum()

def generate_shapes():
    shapes = []
    for mb in range(1, 200):
        shapes.append([mb * 1024 * 256])
        return shapes


def collect_perf(byte_size, cold_boot):
    warm_up_l2_cache_flag = not cold_boot
    wall_time_sum = 0

    getattr(torch, device).empty_cache()

    x = torch.randn(byte_size, dtype=torch.float16, device=device)
    for i in range(10):
        warm_up_l2_cache(x, warm_up_l2_cache_flag)
        my_neg(x)
    torch.accelerator.synchronize()

    if use_profiler:
        with profile(activities=activities) as prof:
            for i in range(100):
                warm_up_l2_cache(x, warm_up_l2_cache_flag)
                my_neg(x)
            torch.accelerator.synchronize()
        kernel_filter = filter(lambda x: "Neg" in x.name or "neg" in x.name, prof.events())
        kernels = list(kernel_filter)
        assert len(kernels) == 100
        for i in range(100):
            assert kernels[i].count == 1
            wall_time_sum += kernels[i].device_time # in us
        return wall_time_sum / 1000


    start = [getattr(torch, device).Event(enable_timing=True) for _ in range(100)]
    end = [getattr(torch, device).Event(enable_timing=True) for _ in range(100)]

    for i in range(100):
        warm_up_l2_cache(x, warm_up_l2_cache_flag)
        start[i].record()

        my_neg(x)
        end[i].record()
    torch.accelerator.synchronize()
    #time.sleep(10)
    for i in range(100):
        wall_time_sum += start[i].elapsed_time(end[i])
    return wall_time_sum


for shape in generate_shapes():
    sum_time_reuse_cache = collect_perf(shape, False)
    size = shape[0] * 2 * 2 / 1024/ 1024
    print((sum_time_reuse_cache/100), "ms")
```

### Versions

main

**Comment by Stonepia** [comment]
Should be oneAPI related. JIRA at CMPLRLLVM-76467

**Comment by Stonepia** [comment]
Impact:

PyTorch team: For PyTorch itself, this only affects torch.Event, so no big impact for PyTorch side.
Customers: for vLLM team, because they rely on torch.Event, so this has more impact for them.

Update:

oneAPI team is WIP on this. Please see the JIRA for update (CMPLRLLVM-76467). No target for oneAPI JIRA for now, so we label it to pytorch 2.15 as pytorch-side track

**Comment by guangyey** [comment]
Update:
The compiler will refine this in 2026.2

**Comment by Stonepia** [comment]
The fix is in https://github.com/intel/llvm/pull/22810
For the oneAPI perspective, starting from internal build `20260806_160000` , and will be public available in 2026.2 , thus we mark the PyTorch 2.15 target still reasonable. So leave this open until from the pytorch side, we upgrade the oneAPI and solve it.

**Comment by Stonepia** [comment]
Move Target to 2.16 since the oneAPI 2026.2 is not upgraded in 2.15
